src/client/editor/inspector.js

```diff
@@ -134,26 +136,34 @@ let inspector = function() {
 
 			// Translation
 			transformFunctions.translate = function(axis, value) {
+				if(isNaN(value)) value = 0;
 				let position = inspector.selected.getPosition();
-				position[axis] = parseFloat(value);
+				position[axis] = value;
 				inspector.selected.setPosition(position);
 			};
 
 			// Scale
 			transformFunctions.scale = function(axis, value) {
+				if(isNaN(value)) value = 1;
 				let scale = inspector.selected.getScale();
-				scale[axis] = parseFloat(value);
+				scale[axis] = value;
 				inspector.selected.setScale(scale);
 			};
 
 			// Rotate
 			transformFunctions.rotate = function() {
 				let rotation = inspector.elements.transform.input.rotate;
+				let x = rotation.x.valueAsNumber;
+				if(isNaN(x)) x = 0;
```

Comment:
Since (from what I've read) `.valueAsNumber` always returns a valid number OR NaN otherwise, wouldn't it be nicer to simply write `let x = rotation.x.valueAsNumber || 0`?